app/controllers/rails_admin/main_controller.rb

```diff
@@ -56,7 +56,14 @@ def respond_to_missing?(sym, include_private)
     end
 
     def back_or_index
-      params[:return_to].presence && params[:return_to].include?(request.host) && (params[:return_to] != request.fullpath) ? params[:return_to] : index_path
+      params[:return_to].presence && allowed_return_to?(params[:return_to]) && (params[:return_to] != request.fullpath) ? params[:return_to] : index_path
```

Comment:
Minor - consolidating the logic for checking if the `return_to` is allowed in the extracted method would make this easier to follow.

```suggestion
      if _allowed_return_to?(params[:return_to].to_s) 
        params[:return_to]
      else
        index_path
      end
```